### PR TITLE
feat: db migration, frontend refactor

### PR DESCRIPTION
**Pull request by yqrashawn**
1. add db migration util
1. db record latest balance only
1. set wallet_addNetwork as internal method
1. refactor frontend code, remove useCurrentAccount, useCurrentNetwork, replace with useCurrentAddress

---
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/helios/310)

### CODE REVIEW

**Comment by angelia-yuqi-personal** [comment]
Task linked: [PORTAL-1666 修复 Bug: dapp approve 无法解析token 字段是空。](https://app.clickup.com/t/3719212/PORTAL-1666)

File: packages/popup/src/hooks/useApi.js [hunk 1]
```diff
@@ -222,30 +214,130 @@ export const useAddressType = address => {
...
+}) => {
+  if (!addressId) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    addressId = useCurrentAddress().data.eid
+  }
+  if (tokenId === '0x0' || tokenId === 'native') {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return useRPC(
+      addressId
+        ? [QUERY_ADDRESS, 'useAddressTokenBalanceSupportNativeToken', addressId]
```
Comment (angelia-yuqi-personal) [inline packages/popup/src/hooks/useApi.js:296]: How about `useNativeTokenBalance` here?

File: packages/popup/src/pages/SendTransaction/components/TokenAndAmount.js [hunk 2]
```diff
@@ -13,50 +13,33 @@ import {
...
-  const {added, native} = useDbHomeAssets()
-  const networkTypeIsCfx = useNetworkTypeIsCfx()
-  const {netId} = useCurrentNetwork()
-  const homeTokenList = [native].concat(added)
+  const {
+    data: {eid: addressId},
+  } = useCurrentAddress()
   const [searchValue, setSearchValue] = useState('')
-  const [tokenList, setTokenList] = useState([...homeTokenList])
-  const onChangeValue = value => {
```
Comment (angelia-yuqi-personal) [inline packages/popup/src/pages/SendTransaction/components/TokenAndAmount.js:32]: why remove this function?

**Review by angelia-yuqi-personal — CHANGES_REQUESTED** [review]

File: packages/popup/src/pages/SendTransaction/components/TokenAndAmount.js (hunk 2 above)
Comment (yqrashawn) [reply packages/popup/src/pages/SendTransaction/components/TokenAndAmount.js:32]: no need to use it any more

File: packages/popup/src/hooks/useApi.js (hunk 1 above)
Comment (yqrashawn) [reply packages/popup/src/hooks/useApi.js:296]: the key indicating it's called inside `useSingleAddressTokenBalanceWithNativeTokenSupport`.
there are too many assumptions here in fallback data and postprocessSuccessData
`useNativeTokenBalance` is too general

File: packages/popup/src/hooks/useApi.js [hunk 3]
```diff
@@ -222,30 +214,130 @@ export const useAddressType = address => {
...
+export const useDbRefetchBalance = param => {
+  useRPC([WALLETDB_REFETCH_BALANCE], param ? {...param} : undefined)
+  useRPC(
+    [WALLETDB_REFETCH_BALANCE, 'REFETCH_ALL'],
+    {type: 'all', allNetwork: true},
+    {refreshInterval: 180000},
+  )
+}
+
+export const useCurrentNetworkToken = ({fuzzy, addressId}) => {
```
[the remark this answers is not in the document]
Comment (yqrashawn) [reply packages/popup/src/hooks/useApi.js:226]: there's no `useCurrentNetworkTokens` it's `useCurrentAddressTokens`

[the remark this answers is not in the document]
Comment (yqrashawn) [reply packages/popup/src/hooks/useApi.js]: I've changed `useCurrentNetworkToken` to `useCurrentNetworkTokens`

[the remark this answers is not in the document]
Comment (angelia-yuqi-personal) [reply packages/popup/src/hooks/useApi.js]: > there's no `useCurrentNetworkTokens` it's `useCurrentAddressTokens`

Did you mean useCurrentAccountTokens? You changed useCurrentAccountTokens to useCurrentAddressTokens

**Review by angelia-yuqi-personal — APPROVED** [review]

**Pull request merged**